### PR TITLE
Story3

### PR DESCRIPTION
**Pull request by JBHarrison5**
https://trello.com/c/cApKb7po/96-story-3-create-a-pattern

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -11,12 +11,12 @@
     <script type="text/javascript" src="script.js" defer></script>
 </head>
 <body>
-  <div id="instructionModal" class="modal">
+<div id="instructionModal" class="modal">
     <div class="modal-content">
-        <div class="introClose">
-            <div></div>
+        <div class="instructionTop">
+            <div class="leftX"></div>
```
Comment (mporam) [inline index.html]: whats this?

File: index.html [hunk 2]
```diff
@@ -26,44 +26,57 @@ <h2>Make a mistake and it's game over!</h2>
         </div>
     </div>
 </div>
+<div id="gameOverModal" class="modal">
+    <div class="modal-content">
+        <div class="gameOverTop">
+            <div class="leftX"></div>
```
Comment (mporam) [inline index.html]: why is there another one??

File: index.html [hunk 3]
```diff
@@ -11,12 +11,12 @@
...
 </head>
 <body>
-  <div id="instructionModal" class="modal">
+<div id="instructionModal" class="modal">
     <div class="modal-content">
-        <div class="introClose">
-            <div></div>
+        <div class="instructionTop">
+            <div class="leftX"></div>
             <h1>Instructions!</h1>
```
Comment (mporam) [inline index.html]: h1 is probably not the right tag to use here. this title isnt the most important title on the page

File: index.html [hunk 4]
```diff
@@ -26,44 +26,57 @@ <h2>Make a mistake and it's game over!</h2>
         </div>
     </div>
 </div>
+<div id="gameOverModal" class="modal">
+    <div class="modal-content">
+        <div class="gameOverTop">
+            <div class="leftX"></div>
+            <h1>Game Over :(</h1>
```
Comment (mporam) [inline index.html]: again, h1 is probably not the right tag here

File: index.html [hunk 5]
```diff
@@ -26,44 +26,57 @@ <h2>Make a mistake and it's game over!</h2>
...
+            <button class="replayButton">Play Again</button>
+        </div>
+    </div>
+</div>
 <div class = 'titles'>
     <div class="gameTitle">Where's the dog been?</div>
     <div class="levelNum">Level X</div>
 </div>
 <div class = 'gameBoxes'>
     <div class = 'row'>
```
Comment (mporam) [inline index.html]: code consistency, why do these have spaces around the `=`? `class = '` dont do this

File: index.html [hunk 6]
```diff
@@ -26,44 +26,57 @@ <h2>Make a mistake and it's game over!</h2>
...
-        <div class="box box2">
+        <div id="box1" class="box box2">
 
         </div>
-        <div class="box box3">
+        <div id="box2" class="box box3">
 
         </div>
     </div>
     <div class = 'row'>
```
Comment (mporam) [inline index.html]: again, incorrect spaces

File: index.html [hunk 7]
```diff
@@ -26,44 +26,57 @@ <h2>Make a mistake and it's game over!</h2>
...
-        <div class="box box5">
+        <div id="box4" class="box box5">
 
         </div>
-        <div class="box box6">
+        <div id="box5" class="box box6">
 
         </div>
     </div>
     <div class = 'row'>
```
Comment (mporam) [inline index.html]: incorrect spaces

File: script.js [hunk 8]
```diff
@@ -1,15 +1,106 @@
...
+    closeModal(instructionModal);
+})
+
+gameOverCloseBtn.addEventListener('click', () => {
+    closeModal(gameOverModal);
+})
+
+const gameOver = () => {
+    openModal(gameOverModal)
+}
```
Comment (mporam) [inline script.js]: better to group function declarations together

File: script.js [hunk 9]
```diff
@@ -1,15 +1,106 @@
...
+// max's get pattern functions:
+
+// const
+let pattern = []
+function getRandBoxes(boxCount) {
```
Comment (mporam) [inline script.js]: why have you changed to using a name function when all the others are constants with fat arrow functions?

File: script.js [hunk 10]
```diff
@@ -1,15 +1,106 @@
...
+    let newArray = [];
```
Comment (mporam) [inline script.js]: what is the purpose of this variable? why not just use the pattern array from the start

File: script.js [hunk 11]
```diff
@@ -1,15 +1,106 @@
...
+
+const box1 = document.querySelector('.box1');
+const box2 = document.querySelector('.box2');
+const box3 = document.querySelector('.box3');
+const box4 = document.querySelector('.box4');
+const box5 = document.querySelector('.box5');
+const box6 = document.querySelector('.box6');
+const box7 = document.querySelector('.box7');
+const box8 = document.querySelector('.box8');
+const box9 = document.querySelector('.box9');
```
Comment (mporam) [inline script.js]: do these actually do anything other than go straight into the array below? if not, why not give them all the same clas sin the html then use `querySelectorAll`, that will give you an array of all boxes in 1 line of code

File: script.js [hunk 12]
```diff
@@ -1,15 +1,106 @@
...
+let boxArray = [box1, box2, box3, box4, box5, box6, box7, box8, box9];
+
 
+const lightDiv = (div, url) => {
+    div.style.backgroundImage = url;
```
Comment (mporam) [inline script.js]: better to do this with a class, so that the CSS controls the styles, not the JS

File: script.js [hunk 13]
```diff
@@ -1,15 +1,106 @@
...
+    setTimeout(() => {
+        div.style.backgroundImage = 'none';
+        div.style.backgroundColor = '';
```
Comment (mporam) [inline script.js]: same here, use a class

File: script.js [hunk 14]
```diff
@@ -1,15 +1,106 @@
...
+    }, 500);
+}
+
+const displayPattern = (pattern) => {
+    for (let i= 0; i<pattern.length; i++) {
```
Comment (mporam) [inline script.js]: incorrect spacing around the operators

Comment (mporam) [reply script.js]: also why not use a forEach on pattern instead of a for loop? tidier syntax

File: script.js [hunk 15]
```diff
@@ -1,15 +1,106 @@
...
+    patternCounter = 0;
+    pattern = [];
+    startGame()
+}
+
+startButton.addEventListener('click', startGame)
+
+let patternCounter = 0;
+
+let boxes = document.querySelectorAll('.box')
```
Comment (mporam) [inline script.js]: surely this can be used instead of `boxArray` above??

File: script.js [hunk 16]
```diff
@@ -1,15 +1,106 @@
...
+for (let box of boxes) {
```
Comment (mporam) [inline script.js]: why not a `forEach`?

File: script.js [hunk 17]
```diff
@@ -1,15 +1,106 @@
...
+    box.addEventListener('click', () => {
+        if (box.id === 'box' + pattern[patternCounter]) {
+            console.log('correct')
```
Comment (mporam) [inline script.js]: remove console.logs please

File: styles.css [hunk 18]
```diff
@@ -55,17 +55,33 @@
     margin: 0.75rem;
 }
 
-.introClose h1 {
+.instructionTop h1 {
     font-size: 40pt;
```
Comment (mporam) [inline styles.css]: dont use pt font sizes, px or rems

File: styles.css [hunk 19]
```diff
@@ -55,17 +55,33 @@
...
+
+.gameOverBtns {
+    display: flex;
+    justify-content: center;
+    align-items: center;
+}
+
+.leftX {
+    width: 30px;
+}
```
Comment (mporam) [inline styles.css]: this class confuses me? it doesnt appear to do anything to do with being on the left or X?

**Review by mporam — CHANGES_REQUESTED** [review]

**Comment by JBHarrison5** [comment]
Latests commit is all the comments resolved, hopefully

File: index.html [hunk 20]
```diff
@@ -11,65 +11,76 @@
...
-        <div class="box box9">
+        <div id="box8" class="box box9">
 
         </div>
     </div>
+
+</div>
+<div class="buttons">
+    <button  class="leaderBoard">Leaderboard</button>
+    <button  class="start">Start</button>
```
Comment (mporam) [inline index.html]: both buttons have 2 spaces before the `class`

**Review by mporam — CHANGES_REQUESTED** [review]

**Review by mporam — APPROVED** [review]

**Pull request merged**